### DISCUSSION TITLE
Knative Support

### DISCUSSION THREAD

**Discussion by borgius** [werf/werf#6180] [General]
Hi Guys!
Thank you amazing product! 
Do you have any plans to support Knative? Looks like It's the next big thing for Kubernetes.

**Comment by shurup** [comment]
Thank you for your feedback! Considering Knative: while we don't have it in our current roadmap, we're always happy to know what our community is willing to have. Can you please shed some light on your case and which kind of support will be useful for that? TIA!

**Comment by ilya-lesikov** [comment]
Better late than never: werf should work with Knative out of the box. Knative doesn't really need any special integration. Just build your Knative functions (OCI images) with werf via Dockerfile or Stapel builder, and then deploy Knative Service Custom Resource via werf.

[Here](https://github.com/knative/docs/blob/main/code-samples/serving/hello-world/helloworld-shell/service.yaml) is an example of Knative Service resource. And [here](https://github.com/knative/docs/tree/a0b1b90eac48642094df3db99f5ef759057c9044/code-samples/serving/hello-world/helloworld-shell) is a complete example, with a Dockerfile, a Service resource and a function. Service resource will go in `.helm/templates/service.yaml` and Dockerfile into the root of repo. Then smack a werf.yaml to the repo root and you mostly good to go.
